src/client/src/scenes/Editor/components/FileButtons/index.tsx

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import { fileButtonData } from './__mock__';
+
+import './styles.scss';
+import { FileButton } from '@app/components/FileButton';
+import { v4 as uuidv4 } from 'uuid';
+import { ExtensionType } from '@app/types';
+import { inject } from 'mobx-react';
+import { EditorDataModel } from '@app/models/editor-data';
+
+interface IFileButtons {
+    editorDataModel?: EditorDataModel;
+}
+
+export const FileButtons: React.FC<IFileButtons> = inject('editorDataModel')(
+    (props) => {
+        const { updateExtension } = props.editorDataModel as EditorDataModel;
```

Comment:
Look to Mobx best practices with React + Typescript. You could write your own hooks, like `useReducer `or `useDispatch`.